flask_restalchemy/resources/utils.py

```diff
@@ -86,13 +86,20 @@ def query_from_request(model, model_serializer, request):
     :param class model:
         SQLAlchemy model class representing a database resource
 
+    :param model_serializer:
+        instance of model serializer
+
     :param request:
         Flask http request data
 
-    :rtype: list
-    :return: the items for the current page
+    :param query:
+        SQLAlchemy query instance
+
+    :rtype: list or dict (pagination)
+    :return: the serialized response with the list or the page of items
     """
-    query = model.query
+    if not query:
```

Comment:
when query is set, check if it is related to the given model

Reply:
As we talked, I found a very tricky way to do this but I do not know exactly what I was restricting.